frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -226,20 +229,58 @@ public void run() {
 
         vw.setWebViewClient(new WebViewClient() {
+            @Override
+            public void onPageFinished(WebView view, String url) {
+                // The url is not correctly updated when navigating to a new page. This allows to
+                // know the current location and to block external requests on that base.
+                view.evaluateJavascript("window.location.pathname", path -> location = path);
+            }
             @Override
             public WebResourceResponse shouldInterceptRequest(final WebView view, WebResourceRequest request) {
-                // Intercept local requests and serve the response from the Android assets folder.
-                try {
+                if (request != null && request.getUrl() != null) {
                     String url = request.getUrl().toString();
-                    InputStream inputStream = getAssets().open(url.replace(BASE_URL, "web/"));
-                    String mimeType = getMimeType(url);
-                    if (mimeType != null) {
-                        return new WebResourceResponse(mimeType, "UTF-8", inputStream);
+                    if (url != null && url.startsWith(BASE_URL)) {
+                        // Intercept local requests and serve the response from the Android assets folder.
+                        try {
+                            InputStream inputStream = getAssets().open(url.replace(BASE_URL, "web/"));
+                            String mimeType = getMimeType(url);
+                            if (mimeType != null) {
+                                return new WebResourceResponse(mimeType, "UTF-8", inputStream);
+                            }
+                            Util.log("Unknown MimeType: " + url);
+                        } catch (IOException e) {
+                            Util.log("Internal resource not found: " + url);
+                        }
+                    } else {
+                        // external request
+                        // allow if location is listed
+                        List<Pattern> patterns = new ArrayList<>();
+                        patterns.add(Pattern.compile("^\"/buy/pocket/.*\"$"));
+                        patterns.add(Pattern.compile("^\"/buy/moonpay/.*\"$"));
+                        patterns.add(Pattern.compile("^\"/account/[^\\/]+/wallet-connect/.*\"$"));
+                        for (Pattern pattern : patterns) {
+                            if (pattern.matcher(location).matches()) {
+                                return super.shouldInterceptRequest(view, request);
+                            }
+                        }
+
+                        String domain = request.getUrl().getHost();
+                        if (domain != null) {
+                            // allow if domain is listed
+                            patterns = new ArrayList<>();
+                            patterns.add(Pattern.compile("^(.*\\.)?verify\\.walletconnect\\.com$"));
```

Comment:
in Qt the pattern looks a bit different https://github.com/digitalbitbox/bitbox-wallet-app/pull/2301/files#diff-c5a5346e0c9e1d112c810b5097daea72abb6bd88406cefb73d52de55c9836f2cR137 - `https://verify\...`. Why is it different here?

Reply:
No reason, I didn't notice that in Qt we were not allowing subdomains. I updated the code here to have the same behavior